File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.go
func findGCD(nums []int) int {
    var mn int = nums[0];
    var mx int = nums[0];
    for _, num := range nums {
        mn = min(mn, num);
        mx = max(mx, num);
    }
    return gcd(mn, mx);
}

func gcd(a, b int) int {
    for b != 0 {
        a, b = b, a % b;
    }
    return a;
}

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findGCD = function(nums) {
    let min = nums[0];
    let max = nums[0];
    for (num of nums) {
        min = Math.min(min, num);
        max = Math.max(max, num);
    }
    return gcd(min, max);
};

function gcd(a, b) {
    while (b !== 0) {
        [a, b] = [b, a % b];
    }
    return a;
}